pkg/apis/k0s.k0sproject.io/v1beta1/kuberouter.go

```diff
@@ -24,19 +24,32 @@ type KubeRouter struct {
 	MTU int `json:"mtu"`
 	// Kube-router metrics server port. Set to 0 to disable metrics  (default: 8080)
 	MetricsPort int `json:"metricsPort"`
-	// Activate Hairpin Mode (allow a Pod behind a Service to communicate to its own ClusterIP:Port)
-	HairpinMode bool `json:"hairpinMode"`
+	// Admits three values: "enabled" enables it globaly, "allowed" allows but services must be annotated explictly and "disabled"
+	Hairpin Hairpin `json:"hairpin"`
+	// DEPRECATED: Use hairpining instead. Activates Hairpin Mode (allow a Pod behind a Service to communicate to its own ClusterIP:Port)
+	HairpinMode bool `json:"hairpinMode,omitempty"`
 	// Comma-separated list of global peer addresses
 	PeerRouterASNs string `json:"peerRouterASNs"`
 	// Comma-separated list of global peer ASNs
 	PeerRouterIPs string `json:"peerRouterIPs"`
 }
 
+type Hairpin string
+
+const (
+	HairpinEnabled  Hairpin = "enabled"
+	HairpinAllowed  Hairpin = "allowed"
+	HairpinDisabled Hairpin = "disabled"
+	// Necessary for backwards compatibility with HairpinMode
+	HairpinUndefined Hairpin = ""
+)
+
```

Comment:
I'd uppercase the values, and I'd add an enum hint for kubebuilder.

```suggestion
// +kubebuilder:validation:Enum=Enabled;Allowed;Disabled
type HairpinMode string

const (
	HairpinEnabled  HairpinMode = "Enabled"
	HairpinAllowed  HairpinMode = "Allowed"
	HairpinDisabled HairpinMode = "Disabled"
	// Necessary for backwards compatibility with HairpinMode
	HairpinUndefined HairpinMode = ""
)

```